bundles/org.openhab.binding.mpd/src/main/java/org/openhab/binding/mpd/internal/handler/MPDHandler.java

```diff
@@ -0,0 +1,289 @@
+        if (command instanceof RefreshType) {
+            handleCommandRefresh(channelUID.getId());
+        } else {
+            handlePlayerCommand(channelUID.getId(), command);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        MPDConfiguration config = getConfigAs(MPDConfiguration.class);
+        connection.start(config.getIpAddress(), config.getPort(), config.getPassword());
+    }
+
+    @Override
+    public void dispose() {
+        ScheduledFuture<?> future = this.futureUpdateStatus;
+        if (future != null && !future.isCancelled()) {
+            future.cancel(true);
+        }
+
+        future = this.futureUpdateCurrentSong;
+        if (future != null && !future.isCancelled()) {
+            future.cancel(true);
+        }
+
+        connection.dispose();
+        super.dispose();
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(MPDActions.class);
+    }
+
+    /**
+     * send a command to the music player daemon
+     *
+     * @param command command to send
+     * @param parameter parameter of command
+     */
+    public void sendCommand(@Nullable String command, String... parameter) {
+        if (command != null) {
+            connection.sendCommand(command, parameter);
+        } else {
+            logger.warn("can't send null command");
+        }
+    }
+
+    private void handleCommandRefresh(String channelId) {
+        stateMap.remove(channelId);
+        switch (channelId) {
+            case CHANNEL_CONTROL:
+            case CHANNEL_STOP:
+            case CHANNEL_VOLUME:
+                scheduleUpdateStatus();
+                break;
+            case CHANNEL_CURRENT_ALBUM:
+            case CHANNEL_CURRENT_ARTIST:
+            case CHANNEL_CURRENT_NAME:
+            case CHANNEL_CURRENT_SONG:
+            case CHANNEL_CURRENT_SONG_ID:
+            case CHANNEL_CURRENT_TITLE:
+            case CHANNEL_CURRENT_TRACK:
+                scheduleUpdateCurrentSong();
+                break;
+        }
+    }
+
+    private synchronized void scheduleUpdateStatus() {
+        logger.debug("scheduleUpdateStatus");
+        ScheduledFuture<?> future = this.futureUpdateStatus;
+        if (future == null || future.isCancelled() || future.isDone()) {
+            this.futureUpdateStatus = scheduler.schedule(() -> doUpdateStatus(), 100, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private void doUpdateStatus() {
+        connection.updateStatus();
+    }
+
+    private synchronized void scheduleUpdateCurrentSong() {
+        logger.debug("scheduleUpdateCurrentSong");
+        ScheduledFuture<?> future = this.futureUpdateCurrentSong;
+        if (future == null || future.isCancelled() || future.isDone()) {
+            this.futureUpdateCurrentSong = scheduler.schedule(() -> doUpdateCurrentSong(), 100, TimeUnit.MILLISECONDS);
```

Comment:
Syntactical sugar. Same for below
```suggestion
            this.futureUpdateStatus = scheduler.schedule(this::doUpdateStatus, 100, TimeUnit.MILLISECONDS);
```

Reply:
Done.